### ISSUE TITLE
Alerts are only generated if another one is being created

### ISSUE THREAD

**Issue by Filisimus** [Graylog2/graylog2-server#13228]
Hi guys,

first I would like to provide some background information. We want to set up a honeypot and as a part of it there is a CMS vulnerable to the Log4Shell exploit. If someone tries to use this exploit a system message is being sent and Graylog should send an email notification to certain people.

So the expected behaviour is that Graylog gets the exploit notification and pretty much immediately sends the email. But instead the following happens (using timestamps for better understanding):

00:00:00 exploit is being used and message with timetamp is saved in logfile
00:00:00 filebeat sends message with timestamp '00:00:00' to Graylog
00:00:00 message can be found via search in Graylog (also with timestamp 00:00:00)

(At this point the event should be triggered and an email being sent but nothing happens)

00:05:00 exploit is being used again
00:05:00 filebeat sends message with timestamp '00:05:00' to Graylog
00:05:00 message can be found via search in Graylog (also with timestamp 00:05:00)
And now the alert for the '00:00:00' appears and the email, which should have been sent at 00:00:00, is being sent now

I can reproduce this easily. I also tried to use different settings for the filter in events but no success. It is probably worth mentioning that this is not related to a single alarm - we have multiple different alerts and this bevahiour also occurs with mixed alarms, e. g. alarm 1 should be triggered but is only triggered after alarm 2 comes up. Lets take a look at 3 screenshots which should explain this better:

![image](https://user-images.githubusercontent.com/55972194/184122046-f4ca310e-dd60-4d6f-bcce-1fdd7edb31dc.png)

Here you can see that the email was received 13:03 but the timestamp is 12:49.

![image](https://user-images.githubusercontent.com/55972194/184122483-712d1484-c9b5-4398-8fd0-d55ea09fcb66.png)

Here is the next email I received, arrival was 13:08 but the timestamp is 13:03

Here is the overview for the alerts. You will also notice that the alert for 13:08:08 should have been a MySQL message but in the email it says Log4J exploit:

![image](https://user-images.githubusercontent.com/55972194/184125068-070b35c3-1ec2-4caa-9532-2f2fe4dacd3a.png)

Also the email delivery is not the issue - if an alert is being generated the email is being sent immediately. 

![image](https://user-images.githubusercontent.com/55972194/184123028-7d201c42-431e-4b2b-a56a-b2becc7ac736.png)

Here you have my filter settings. 

TLDR:
Message appears in Graylog search but the alert (and so the email) are only being triggered after a new alert is triggered again.

* Graylog Version: Graylog v4.3.4
* Java Version: 1.8.0_342
* Elasticsearch Version: 7.10.2
* MongoDB Version: 4.0.28
* Operating System: Ubuntu 20.04
* Browser version: Chrome 104.0.5112.81

I appreciate any help. :)

Cheers, Fili

**Comment by HenryTheSir** [comment]
Hi,

is the message for the second alert the first processed message after the one which should be alerted?

best regards

**Comment by Filisimus** [comment]
My apologies, I dont know what exactly you are asking. The messages itself are correct for each email generated, so there is no mix up between them.

**Comment by Filisimus** [comment]
Hi guys,

maybe my first posting was too confusing, I'm sorry if that's the case. I decided to provide a detailed list how to reproduce the issue:

1) Download and install VMWare Workstation, download Ubuntu Server 20.04 from https://releases.ubuntu.com/20.04/ubuntu-20.04.4-live-server-amd64.iso, Create a New virtual machine, use the just downloaded ISO as Installer disk image, assign 30 GB disk space, 4 GM RAM and 4 CPUs
2) Start Virtual machine and the Ubuntu installation will start. Follow the instructions to install it. I recommend to  install OpenSSH when asked. Finally Ubuntu will be installed, this might take some time. Wait until it asks you to reboot your server at the bottom of the window.
3) After rebooting Ubuntu connect with your favorite SSH client (e. g. Putty) to your server. If you dont know the IP use the command 'ip a' in the VMWare Workstation window where the Ubuntu server is running. You could also use VMWare Workstation alone but using Putty or any other SSH Client allows you to copy / paste the following commands much easier.
4) The next steps are taken from the official Graylog installation guide found here: https://docs.graylog.org/docs/ubuntu I will repeat them for the sake of clarity:
  - sudo apt-get update && sudo apt-get upgrade -y
  - sudo apt install -y apt-transport-https openjdk-8-jre-headless uuid-runtime pwgen
  - sudo apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 9DA31620334BD75D9DCB49F368818C72E52529D4
  - echo "deb [ arch=amd64 ] https://repo.mongodb.org/apt/ubuntu bionic/mongodb-org/4.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-4.0.list
  - sudo apt-get update
  - sudo apt-get install -y mongodb-org
  - wget -q https://artifacts.elastic.co/GPG-KEY-elasticsearch -O myKey
  - sudo apt-key add myKey
  - echo "deb https://artifacts.elastic.co/packages/oss-7.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-7.x.list
  - sudo apt-get update && sudo apt-get install elasticsearch-oss
  - wget https://packages.graylog2.org/repo/packages/graylog-4.3-repository_latest.deb
  - sudo dpkg -i graylog-4.3-repository_latest.deb
  - sudo apt-get update && sudo apt-get install graylog-server graylog-enterprise-plugins graylog-integrations-plugins graylog-enterprise-integrations-plugins
  - Execute this command and choose your password for your Graylog server - save the output it in a textfile, we will need it soon: echo -n "Enter Password: " && head -1 </dev/stdin | tr -d '\n' | sha256sum | cut -d" " -f1
  - Run this command and again save the output in a textfile: pwgen -N 1 -s 96
  - sudo vi /etc/graylog/server/server.conf
  -- Look for 'password_secret = ' and enter the out of the 'pwgen -N 1 -s 96' command
  -- The next entry is 'root_password_sha = ' enter here the output of the password command 
  -- Uncomment (remove the '#' at the beginning of the line) the entry '#http_bind_address = 127.0.0.1:9000' and replace the IP with your server IP.
  -- save and exit file
  - sudo vi /etc/elasticsearch/elasticsearch.yml
  -- uncomment and change: cluster.name: graylog
  -- add below this: action.auto_create_index: false
  -- save and exit file
  - sudo systemctl daemon-reload
  - sudo systemctl enable mongod.service
  - sudo systemctl enable elasticsearch.service
  - sudo systemctl enable graylog-server.service
  - sudo systemctl restart mongod.service
  - sudo systemctl restart elasticsearch.service
  - sudo systemctl start graylog-server.service
  
5) Open http://YOUR-SERVER-IP:9000/ and you should be greeted by the Graylog login window. Enter 'admin' as username and the password you chose earlier
6) Go to 'System / Overview' -> 'Inputs', choose 'Beats' from the 'Select input' dropdown list. Click on 'Launch new Input' right next to it. Assign a title you like and leave everything else as it is. Click on 'save' to save it.
7) Now we need something to generate logfiles to feed Graylog, I decided to go here with Apache2.
8) sudo apt install apache2 -y
9) cd /var/www/html
10) sudo touch test.html
11) Next we require Filebeat to transfer the Apache2 logfiles to Graylog
12) wget -qO - https://artifacts.elastic.co/GPG-KEY-elasticsearch | sudo apt-key add -
13) sudo apt-get install apt-transport-https
14) echo "deb https://artifacts.elastic.co/packages/8.x/apt stable main" | sudo tee -a /etc/apt/sources.list.d/elastic-8.x.list
15) sudo apt-get update && sudo apt-get install filebeat
16) sudo systemctl enable filebeat
17) sudo vi /etc/filebeat/filebeat.yml
  - set 'enabled:' from 'false' to 'true'
  - replace the entry '- /var/log/*.log' with '- /var/log/apache2/access.log' (without the 's)
  - uncomment 'include_lines' and replace " ['^ERR', '^WARN']" with "['test.html']" (without the " but with ' this time)
  - comment 'output.elasticsearch'
  - comment 'hosts: ["localhost:9200"]' below
  - uncomment 'output.logstash'
  - uncomment  'hosts' below, enter your server IP
  - save and quit
18) Now we need the event, go to http://YOUR-SERVER-IP:9000/alerts and click on 'Get Started!'  
  - Choose a descriptive Title and Description, click 'Next'
  - For 'Condition Type' choose 'Filter & Aggregation', click 'Next'
  - Click 'Next'
  - Click 'Next'
  - Click 'Done'
19) sudo systemctl start filebeat
20) Open http://YOUR-SERVER-IP/test.html
21) Go to http://YOUR-SERVER-IP:9000/search and wait a few seconds. Refresh and you should see 1 entry.
22) Go to http://YOUR-SERVER-IP:9000/alerts and you will see that no alert or event has been triggered.
23) Wait 5 minutes
24) Open http://YOUR-SERVER-IP/test.html again 
25) Go back to http://YOUR-SERVER-IP:9000/search, expand the timeframe by changing 'From' at the top of the page to '5 hours ago'. You will see 2 entries now, the one we just created and the one 5 minutes ago.
26) Go to http://YOUR-SERVER-IP:9000/alerts and you will see that an event has been triggered now, but with the timestamp 5 minutes ago.
 
So... first thanks for staying with me. Could anyone explain to me this odd behaviour, please? Or is it a bug? Because I would expect that every entry triggers the event, so there should have been 2 events?
 
Cheers,
Fili

**Comment by Filisimus** [comment]
Hiya,

well that sucks but explains why this is happening, thanks a lot. I will find a way to create a steady stream for the events to trigger.

Cheers,
Fili
